### PR TITLE
Update react-native-test-app to pick up latest security fixes

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [ ] win32 (Office)
- [x] windows
- [x] android

### Description of changes

(Apple only) RNTA version 1.1.4 updates it's xcconfigs so that the Address Sanitizer (ASAN) and Udefined Behavior Sanitizer (UBSAN) are OFF for release builds. We need this for downstream consumption. By updating our dependency, we can also remove one of our xcconfig flags where we hardcoded ASAN as off for release.

### Verification

CI 

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Review by tido64 — APPROVED** [review]

**Comment by harrieshin** [comment]
should we still leave the xcconfig to future proof any changes in react native test app?

**Comment by Saadnajmi** [comment]
> should we still leave the xcconfig to future proof any changes in react native test app? 

That's an interesting perspective. I guess there's no problem with duplicate xcconfigs and it's better to be explicit what this project requires? In that case I'll add the flag to disable UBSAN as well

File: .ado/xcconfig/publish_overrides.xcconfig [hunk 1]
```diff
@@ -13,5 +13,8 @@ ONLY_ACTIVE_ARCH=NO
 // Specify the exact Swift version used for reproducibility
 SWIFT_VERSION = 5.0
 
-// Turn off ASAN for ship builds
-ENABLE_ADDRESS_SANITIZER = NO
+// Turn off Sanitizers for Release Builds
+CLANG_ADDRESS_SANITIZER = NO
+CLANG_UNDEFINED_BEHAVIOR_SANITIZER = NO
+
```
Comment (Saadnajmi) [inline .ado/xcconfig/publish_overrides.xcconfig:19]: Feels weird to explicitly set it to no when the default is NO, @harrieshin

Comment (harrieshin) [reply .ado/xcconfig/publish_overrides.xcconfig:19]: it shouldn't have any effect on build tho.

**Review by harrieshin — APPROVED** [review]

**Pull request merged**